.github/workflows/main.yml

```diff
@@ -0,0 +1,75 @@
+name: Build and Upload JAR
+
+on:
+  push:
+    branches:
+      - master
+    paths:
+      - 'src/**'
+      - 'build.gradle'
+      - '.github/workflows/main.yml'
+  pull_request:
+    branches:
+      - '*'
+    paths:
+      - 'src/**'
+      - 'build.gradle'
+      - '.github/workflows/main.yml'
+
+env:
+  GRADLE_OPTS: -Dorg.gradle.daemon=false
+  DO_NOT_TRACK: true
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up JDK 17
```

Comment:
Is there a particular reason to use JDK 17? If no, I would prefer JDK 11

Reply:
The toolchain in build.gradle specifies the java version as 17. Wanted to be consistent with that. Should I change it to JDK 11 in the build.gradle too?
https://github.com/wherobots/wherobots-jdbc-driver/blob/ee16d5b666f109038b4f0f6a7297120c13b38f86/lib/build.gradle#L38

Reply:
@mpetazzoni can you clarify on this? Is Java 17 really needed?

Reply:
CC @peterfoldes

Reply:
There shouldn't be a strict need for >= 17, so we could drop down to >= 11.

Note that 11 is on its way out of support though.